### DISCUSSION TITLE
Setting the 'Content-Disposition' header for arbitrary content

### DISCUSSION THREAD

**Discussion by ee-usgs** [micronaut-projects/micronaut-core#9463] [Q&A]
I am streaming CSV data from a Controller method - How do I set the `Content-Disposition: attachment` header so that browsers download it instead of display it?

I see that for sending 'Files', [this is done automatically](https://docs.micronaut.io/latest/guide/#transfers), but I don't intend to use either of the two return types shown in that example, my method signature looks like this:

```java
@Get("/result{?values*}")
public Flux<String> result(@Nullable @QueryValue("values") Map<String, String> filterMap);
```

Isn't there a way to just annotate the content disposition?  It seems weird to me that it currently seems tied to specific response types - In general, any response of any type might be inline or attachment.  It's an independent concern.

Thank you in advance for the help.

--ee

**Comment by yawkat** [comment]
You can set the header manually by returning a HttpResponse.

**Comment by ee-usgs** [comment]
@yawkat If I return a `HttpResponse`, I break all of Micronaut's automatic handling of `Flux`.
